nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiWebConfigurationContext.java

```diff
@@ -452,8 +456,40 @@ private ComponentDetails getComponentConfiguration(final ProcessorDTO processor)
                     .state(processor.getState())
                     .annotationData(processorConfig.getAnnotationData())
                     .properties(processorConfig.getProperties())
+                    .descriptors(getComponentDescriptors(processorConfig))
                     .validateErrors(processor.getValidationErrors()).build();
         }
+
+        private Map<String,ComponentDescriptor> getComponentDescriptors(final ProcessorConfigDTO processorConfig){
```

Comment:
Do you think it would be more appropriate to name the operation _buildComponentDescriptors(..)_ instead of _get*_?
IMHO _get*_ usually implies an operation that is returning something that already exist. Your call, but wanted to share.